src/Classes/MapController.h

```diff
@@ -6,27 +6,32 @@
 
 #include "IMapEventListener.h"
 namespace tsg {
-    namespace map {
-        class MapController{
-            std::string mapsRoot;
-            std::vector<IMapEventListener*> mapEventListeners;
-            cocos2d::TMXTiledMap *currentMap;
-            typedef std::tuple<std::string, std::string> MapType;
-            typedef std::vector<MapType> MapCollection;
-            void notifyListeners();
-            
-        protected:
-            
-        public:
-            virtual void loadMapFromFile(const std::string&);
-            MapController(const MapController &);
-            MapController();
-            MapController(std::string);
-            MapCollection getMaps();
-            void loadMap(std::string);
-            void registerListener(IMapEventListener*); 
-        };
-    }
+namespace map {
+class MapController {
+
+  std::string mapsRoot;
+  std::vector<IMapEventListener *> mapEventListeners;
+  cocos2d::TMXTiledMap *currentMap;
+  cocos2d::EventListenerTouchOneByOne *listener;
+  cocos2d::Vec2 touchPositionStarted;
+  cocos2d::Layer *gameLayer;
+  typedef std::tuple<std::string, std::string> MapType;
+  typedef std::vector<MapType> MapCollection;
+  void notifyListeners();
+  void initTouchEvents();
+
+ protected:
+ public:
+  virtual void loadMapFromFile(const std::string &);
+
+  MapController(const MapController &);
+  MapController(cocos2d::Layer*);
+  MapController(cocos2d::Layer*, std::string);
+
+  MapCollection getMaps();
+  void loadMap(std::string);
+  void registerListener(IMapEventListener *);
+};
```

Comment:
Actually I made them public as part of public interface. Except of `loadMapFromFile`. It needed for mocking. I'll look for possibility to make it private.